aii-opennebula/src/main/perl/opennebula.pm

```diff
@@ -180,16 +192,75 @@ sub new
     return bless {}, $class;
 }
 
+sub change_permissions
+{
+    my ($self, $one, $type, $resource, $permissions) = @_;
+    my ($method, $id, $instance, %chown, $out);
+    my @options = qw(user group mode);
+    # -1 value by default, owner and group are not changed
+    %chown = (
+        uid => -1,
+        gid => -1,
+        one => $one,
+    );
+
+    foreach my $value (@options) {
+        if (exists($permissions->{$value})) {
+            if ($value eq "mode") {
+                $out = $resource->chmod($permissions->{$value});
+                if ($out) {
+                    $main::this_app->info("Changed $type mode id $out to: ", $permissions->{$value});
+                } else {
+                    $main::this_app->error("Not able to change $type mode to: ", $permissions->{$value});
+                };
+            } else {
+                if ($value eq "user") {
+                    $chown{uid} = $permissions->{$value};
+                } else {
+                    $chown{gid} = $permissions->{$value};
+                };
+            };
+        };
+    };
+
+    $out = $resource->chown(%chown);
+    if ($out) {
```

Comment:
use a `my $msg = "user:group $chown{uid}:$chown{gid} for: " . $resource->name;` for th ecommon message part